src/java.net.http/share/classes/jdk/internal/net/http/Stream.java

```diff
@@ -137,6 +137,7 @@
     private volatile boolean remotelyClosed;
     private volatile boolean closed;
     private volatile boolean endStreamSent;
+    private volatile boolean finalResponseCodeReceived = false;
```

Comment:
Hello Conor, I've seen some efforts/PRs in recent times where explicit assignment of the type's default value to a `volatile` field was removed for performance considerations. So, maybe this field can just be:
```
private volatile boolean finalResponseCodeReceived;
```

Reply:
True yes, in fact this has been done elsewhere in the PR actually for `PushedStream.finalPushResponseCodeReceived` where it was declared as you suggested for the same reasons. I will update as you have suggested.